intelmq/bots/collectors/rsync/collector_rsync.py

```diff
@@ -23,20 +42,73 @@ def init(self):
+            try:
+                mkdir(self.privkeydir)
+            except FileExistsError:
+                pass
+            self.private_key_path = path.join(self.privkeydir, 'private.key') 
+
+            # privkey format parser, support formats with and without headers, breaklines, etc.
+            if '-----' in self.private_key:
+                p = self.private_key.split('-----')[2]
+            else:
+                p = self.private_key
+            pb64 = ''.join(p.split())
+            fullkey = ['-----BEGIN OPENSSH PRIVATE KEY-----']
+            fullkey += [ pb64[i:i+70] for i in range(0, len(pb64), 70) ]
+            fullkey += ['-----END OPENSSH PRIVATE KEY-----']
+            fullkey += ['']
+            final_key = '\n'.join(fullkey)
+
+            with open(self.private_key_path, 'w') as f:
+                f.write(final_key)
+            chmod(self.private_key_path, stat.S_IRUSR | stat.S_IWUSR)
+
+        if self.private_key_path:
+            self.strict_host_key_checking_str = 'yes' if self.strict_host_key_checking else 'no'
+            self.extra_params += ['-e', f'ssh -i {self.private_key_path} -o StrictHostKeyChecking={self.strict_host_key_checking_str}']
+
     def process(self):
-        self.logger.info(f"Updating file {self.file}.")
-        process = run(["rsync", path.join(self.rsync_path, self.file),
-                       self.temp_directory],
-                      stderr=PIPE)
-        if process.returncode != 0:
-            raise ValueError("Rsync on file {!r} failed with exitcode {} and stderr {!r}."
-                             "".format(self.file,
-                                       process.returncode,
-                                       process.stderr))
-        report = self.new_report()
-        with open(path.join(self.temp_directory, self.file)) as rsync_file:
-            report.add("raw", rsync_file.read())
-            self.send_message(report)
+        formatting = self.rsync_file_path_formatting
+        rsync_file = self.file
+        rsync_path = self.rsync_path
+        if formatting:
+            try:
+                rsync_file = rsync_file.format(time=Time(formatting))
+                rsync_path = rsync_path.format(time=Time(formatting))
+            except TypeError:
+                self.logger.error(f"Wrongly formatted rsync_file_path_formatting parameter: {formatting}. Should be boolean (False) or a time-delta JSON.")
+                raise
+            except KeyError:
+                self.logger.error(f"Wrongly formatted file '{rsync_file}' or rsync_path '{rsync_path}'. Possible misspell with 'time' on 'formatting' variable.")
+                raise
+        rsync_full_path = path.join(rsync_path, rsync_file)
+
+        self.logger.info(f"Updating file {rsync_file}.")
+        cmd_list = ["rsync"] + self.extra_params + [rsync_full_path, self.temp_directory]
+        self.logger.info(f"CMD: {cmd_list}")
```

Comment:
```suggestion
        self.logger.debug(f"Executing command: {cmd_list}.")
```

Reply:
Hi @sebix. I made the changes in log messages, but the tests still failing. What am I not seeing?